### PR TITLE
[G-690] | Add basic HIPLinkInit interface

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java  1
- src/main/java/in/projecteka/consentmanager/hipLinkInitialization/UserAuthController.java  1
- src/main/java/in/projecteka/consentmanager/hipLinkInitialization/UserAuthService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5914384&bid=19378147)

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package in.projecteka.consentmanager.hipLinkInitialization;
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java:1]: IMO, this should not be a top level package. 
I think it should be a package under consentmanager/link/

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import static in.projecteka.consentmanager.common.Constants.CURRENT_VERSION;
+
+public class Constants {
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java:5]: If you move the package under consentmanager/link, there is already a constant file available there

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/UserAuthService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package in.projecteka.consentmanager.hipLinkInitialization;
+
+import in.projecteka.consentmanager.hipLinkInitialization.model.request.AuthInitRequest;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+@AllArgsConstructor
+public class UserAuthService implements HIPLinkInitializer {
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/hipLinkInitialization/UserAuthService.java:10]: I would have thought that the UserAuthService will always be there, but specific implementation of HIPLInkInit may not be there. So UserAuthService should be composed of instances of HIPLinkInitializer

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/HIPLinkInitializer.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package in.projecteka.consentmanager.hipLinkInitialization;
+
+import in.projecteka.consentmanager.hipLinkInitialization.model.request.AuthInitRequest;
+import reactor.core.publisher.Mono;
+
+public interface HIPLinkInitializer {
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/hipLinkInitialization/HIPLinkInitializer.java:6]: Can we think of better name? Its specific to an action .. so maybe HIPLinkInitAction or so?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java  1
- src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthController.java  1
- src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthInitAction.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5914384&bid=19378147)

File: src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+@AllArgsConstructor
+public class UserAuthentication {
+    HIPLinkInitAction initAction;
+    private final Logger logger = LoggerFactory.getLogger(UserAuthentication.class);
+
+    public Mono<Void> authInit(AuthInitRequest request) {
+        logger.info("User auth initialization");
+        initAction.execute(request);
+        return null;
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java:17]: please return empty mono.

File: src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/AuthInitRequest.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package in.projecteka.consentmanager.link.hip_link.model.request;
+
+import lombok.Value;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Value
+public class AuthInitRequest {
+    UUID requestId;
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/AuthInitRequest.java:10]: This is not UUID but a string. We are advising the UUID format, but need not be so, its string in the contract.

File: src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/Requester.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package in.projecteka.consentmanager.link.hip_link.model.request;
+
+import lombok.Value;
+
+@Value
+public class Requester {
+    Integer id;
+    String type;
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/Requester.java:8]: You can make this ENUM - there are only 2 types of requesters. You can just start with HIP (as the use case is only for Link right now)

File: src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/AuthInitRequest.java (hunk 6 above)
Comment (ganesanarun) [reply src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/AuthInitRequest.java:10]: but requestIds are treated as UUID only, even in gateway, 

are you saying only for this or everything?

File: src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/Requester.java (hunk 7 above)
Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/Requester.java]: done

File: src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java (hunk 5 above)
Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java]: done

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java (hunk 1 above)
Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java]: done

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java (hunk 2 above)
Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/hipLinkInitialization/Constants.java]: done

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/UserAuthService.java (hunk 3 above)
Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/hipLinkInitialization/UserAuthService.java]: done

File: src/main/java/in/projecteka/consentmanager/hipLinkInitialization/HIPLinkInitializer.java (hunk 4 above)
Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/hipLinkInitialization/HIPLinkInitializer.java]: done

File: src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/AuthInitRequest.java (hunk 6 above)
Comment (angshu) [reply src/main/java/in/projecteka/consentmanager/link/hip_link/model/request/AuthInitRequest.java:10]: I think we should do that everywhere

File: src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import reactor.core.publisher.Mono;
+
+@AllArgsConstructor
+public class UserAuthentication {
+    HIPLinkInitAction initAction;
+    private final Logger logger = LoggerFactory.getLogger(UserAuthentication.class);
+
+    public Mono<Void> authInit(AuthInitRequest request) {
+        logger.info("User auth initialization");
+        initAction.execute(request);
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java:16]: you can just return over here.
return initAction.execute()

Comment (MeghnaThoughtworks) [reply src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java]: done

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthController.java  1
- src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthentication.java  1
- src/main/java/in/projecteka/consentmanager/link/hip_link/UserAuthInitAction.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5914384&bid=19378147)

**Pull request merged**